maintenance/projects/js-toolkit/packages/liferay-npm-build-tools-common/src/namespace.ts

```diff
@@ -109,5 +109,9 @@ export function getNamespace(moduleName: string): string {
  * @return the namespace for modules
  */
 export function makeNamespace({name}: {name: string}): string {
+	if (name.startsWith('@')) {
+		name = name.substr(1).replace('/', '!');
```

Comment:
>  My impression is that we could allow separators here in a non ambiguous way,  but it would require a cascade of changes in other parts of the system

Yes and no. We could change the scheme spending a lot of money and time in our code (with backports up to 7.1). But then, we would break customer's code, because we had to do these to maintain a backward compatible migration path from the very first version of AMD loader (this was even before the bundler existed).

So, if we wanted to use `/`s we would have to either break old code or support two different schemes in newer versions :scream: . I would like to explain it better to you but this stuff is old and I would have to investigate and re-read all the blog posts I wrote to be able to explain all the details :-(. However, I'm pretty sure that I introduced the `!` notation for those reasons. We also use `%2F` in place of `/` inside the JAR files for the very same reason.

Reply:
> in the event that we had more than `/`

We could, but it would mean that a `package.json` file is very badly broken because this method is fed parsed `package.json` files. 

It wouldn't hurt to check for additional `/`s, of course, but there are so many things that can be incorrect inside a `package.json` (given that it is technically "user input") that we cannot check for every possibility in advance. I mean, it could even contain emoji symbols :sweat_smile: .

Reply:
Seems fair to ignore this possibility then. Given that we are going to lint that all of our OSGi packages use `@liferay/`-prefixed names, we can lint for extra `/` at the same time, and this code can just assume the names are good.

Reply:
I added this commentary https://github.com/liferay/liferay-frontend-projects/pull/408/commits/e8a8443a7fe96d07e74aeaec76a8fb0274d069f1